packages/cli/test/integration/workflows.controller.ee.test.ts

```diff
@@ -0,0 +1,125 @@
+import express from 'express';
+
+import * as utils from './shared/utils';
+import * as testDb from './shared/testDb';
+import { createWorkflow } from './shared/testDb';
+import * as UserManagementHelpers from '../../src/UserManagement/UserManagementHelper';
+import { v4 as uuid } from 'uuid';
+
+import type { Role } from '../../src/databases/entities/Role';
+import config from '../../config';
+
+jest.mock('../../src/telemetry');
+
+// mock whether sharing is enabled or not
+jest.spyOn(UserManagementHelpers, 'isSharingEnabled').mockReturnValue(true);
+
+let app: express.Application;
+let testDbName = '';
+
+let globalOwnerRole: Role;
+let globalMemberRole: Role;
+
+beforeAll(async () => {
+	app = await utils.initTestServer({
+		endpointGroups: ['workflows'],
+		applyAuth: true,
+	});
+	const initResult = await testDb.init();
+	testDbName = initResult.testDbName;
+
+	globalOwnerRole = await testDb.getGlobalOwnerRole();
+	globalMemberRole = await testDb.getGlobalMemberRole();
+
+	utils.initConfigFile();
+
+	utils.initTestLogger();
+	utils.initTestTelemetry();
+
+	config.set('enterprise.workflowSharingEnabled', true);
+});
+
+beforeEach(async () => {
+	await testDb.truncate(['User', 'Workflow', 'SharedWorkflow'], testDbName);
+});
+
+afterAll(async () => {
+	await testDb.terminate(testDbName);
+});
+
+test('PUT /workflows/:id/share should save sharing with new users', async () => {
+	const owner = await testDb.createUser({ globalRole: globalOwnerRole });
+	const member = await testDb.createUser({ globalRole: globalMemberRole });
+	const workflow = await createWorkflow({}, owner);
+
+	const authOwnerAgent = utils.createAgent(app, { auth: true, user: owner });
```

Comment:
You can also use the shorter syntax by initializing the `authAgent` function, as in `credentials.ee.test.ts`.